src/lib.rs

```diff
@@ -3,12 +3,13 @@ use std::ops;
 use std::ops::Mul;
 use std::ops::{Index, IndexMut};
 
+///An abstraction over a 2D matrix of f64s. the foundation of this library
```

Comment:
```suggestion
/// An abstraction over a 2D matrix of f64s. the foundation of this library
```
nit, a space is more readable